internal/cmd/eventgen/eventtypes.go

```diff
@@ -89,17 +91,30 @@ func generateTypes(schemaDefs map[string][]schemaDefinitionRenderer, outputDir s
 	return nil
 }
 
-// latestMajorVersions inspects a list of schema definitions for a single type and
-// maps each encountered major version to the schemaDefinitionRenderer that
-// represents the most recent minor.patch version within that major version.
-func latestMajorVersions(schemas []schemaDefinitionRenderer) map[int64]schemaDefinitionRenderer {
-	majorVersions := map[int64]schemaDefinitionRenderer{}
+// significantVersions inspects a list of schema definitions for a single type and maps the type
+// versions to generate to the corresponding schemaDefinitionRenderers. For non-experimental
+// event versions, the type version will be the major version of the most recent major.minor.patch
+// version within each major version will be returned, while all experimental event versions will
+// be returned with the type version set to the full major.minor.patch version.
+//
+// For example, given the versions (0.1.0, 0.2.0, 1.0.0, 1.1.0, 2.0.0) a map with the keys
+// (0.1.0, 0.2.0, 1, 2) will be returned.
+func significantVersions(schemas []schemaDefinitionRenderer) map[string]schemaDefinitionRenderer {
+	versions := map[string]schemaDefinitionRenderer{}
 	for _, schema := range schemas {
-		if current, exists := majorVersions[schema.Version().Major()]; !exists || current.Version().LessThan(schema.Version()) {
-			majorVersions[schema.Version().Major()] = schema
+		// Keep all schemas for experimental versions.
+		if schema.Version().Major() == 0 {
+			versions[schema.Version().String()] = schema
+			continue
+		}
+
+		// Only keep the latest non-experimental version.
+		majorStr := strconv.Itoa(int(schema.Version().Major()))
```

Comment:
Why do you need to do this "cast to int -> convert to string" method here when you could just do the .String() before?

Reply:
Because in the experimental case we want the significant version (the map key) to be the full version but in the other cases we just want it to be the major version. See the example on lines 100–101. The keys of the map we return are the versions to generate.

Reply:
Ok, followed the code to the semver dependency and understand why you need to cast it now.